packages/remix-node/form-data.ts

```diff
@@ -1,32 +1,144 @@
+import { File } from "web-file-polyfill";
+import Busboy from "busboy";
+import type { Readable } from "stream";
+
+export type UploadHandlerArgs = {
+  name: string;
+  stream: Readable;
+  filename: string;
+  encoding: string;
+  mimetype: string;
+};
+
+export type UploadHandler = (
+  args: UploadHandlerArgs
+) => Promise<string | File | undefined>;
+
+export async function parseFormData(
+  contentType: string,
+  stream: Readable,
+  abortController?: AbortController,
+  uploadHandler?: UploadHandler
+) {
+  let fields: Record<string, (File | string)[]> = {};
+  let fileWorkQueue: Promise<void>[] = [];
+
+  await new Promise<void>(async (resolve, reject) => {
+    let busboy = new Busboy({
+      highWaterMark: 2 * 1024 * 1024,
+      headers: {
+        "content-type": contentType
+      }
+    });
+
+    let aborted = false;
+    function abort(error?: Error) {
+      if (aborted) return;
+      aborted = true;
+
+      stream.unpipe();
+      stream.removeAllListeners();
+      busboy.removeAllListeners();
+
+      abortController?.abort();
+      reject(error || new Error("failed to parse form data"));
+    }
+
+    busboy.on("field", (name, value) => {
+      fields[name] = fields[name] || [];
+      fields[name].push(value);
+    });
+
+    if (uploadHandler) {
```

Comment:
We need to warn them when they don't have an upload handler and the request body contains files.